crates/miden-testing/src/mock_chain/chain.rs

```diff
@@ -559,12 +575,20 @@ impl MockChain {
     /// a chain of transactions against the same account that build on top of each other. For
     /// example, transaction A modifies an account from state 0 to 1, and transaction B modifies
     /// it from state 1 to 2.
-    pub fn build_tx_context(
+    pub fn build_tx_context_at_block(
```

Comment:
```suggestion
    pub fn build_tx_context_at(
```

Nit: Maybe this would be sufficient. We use a similar approach for `MockChain::prove_next_block_at`, `ProposedBlock::new_at`, etc. so might be more consistent.

Similarly for `get_transaction_inputs_at`.